modules/imap_folders/modules.php

```diff
@@ -140,18 +140,25 @@ public function process() {
             }
 
             $specials = $this->user_config->get('special_imap_folders', array());
+            
             if ($exposed = $imap->get_special_use_mailboxes()) {
-                $specials[$form['imap_server_id']] = array('sent' => $exposed['sent'], 'draft' => '', 'trash' => $exposed['trash'], 'archive' => '', 'junk' => '');
+                $specials[$form['imap_server_id']] = array(
+                    'sent' => $exposed['sent'] ?? '',
+                    'draft' => $exposed['drafts'] ?? '',
+                    'trash' => $exposed['trash'] ?? '',
+                    'archive' => $exposed['archive'] ?? '',
+                    'junk' => $exposed['junk'] ?? ''
+                );
             } else if ($form['imap_service_name'] == 'gandi') {
                 $specials[$form['imap_server_id']] = array('sent' => 'Sent', 'draft' => 'Drafts', 'trash' => 'Trash', 'archive' => 'Archive', 'junk' => 'Junk');
             } else {
                 $specials[$form['imap_server_id']] = array('sent' => '', 'draft' => '', 'trash' => '', 'archive' => '', 'junk' => '');
-            }     
+            }
             $this->user_config->set('special_imap_folders', $specials);
 
             $user_data = $this->user_config->dump();
             $this->session->set('user_data', $user_data);
-            
+                        
```

Comment:
Please be careful with whitespace - we don't want lines containing trailing whitespaces.

Reply:
noted